packages/ember-data/lib/system/relationships/state/relationship.js

```diff
@@ -228,12 +228,19 @@ Relationship.prototype = {
     var self = this;
     //TODO Once we have adapter support, we need to handle updated and canonical changes
     self.computeChanges(records);
+    self.setHasData(true);
   },
 
   notifyRecordRelationshipAdded: Ember.K,
   notifyRecordRelationshipRemoved: Ember.K,
 
-  setHasData: function() {
+  setHasData: function(value) {
+    var inverseKey = this.inverseKey;
+    if (inverseKey) {
+      this.canonicalMembers.forEach(function(member) {
```

Comment:
This breaks if your inverse is a many relationship, because even though you only added one record, you are saying it has all of them. Not really sure what to do about this though. Also this seems like the wrong place, probably should happen automatically as the setHasData gets called by addCanonicalRecord